library/psa_crypto.c

```diff
@@ -6360,4 +6373,270 @@ psa_status_t psa_crypto_init( void )
     return( status );
 }
 
+psa_status_t psa_pake_setup(
+    psa_pake_operation_t *operation,
+    const psa_pake_cipher_suite_t *cipher_suite )
+{
+    if( operation->stage != PSA_PAKE_OPERATION_STAGE_COLLECT_INPUTS )
+        return( PSA_ERROR_BAD_STATE );
+
+    if( operation->data.inputs.alg != PSA_ALG_NONE )
+        return( PSA_ERROR_BAD_STATE );
+
+    if( cipher_suite == NULL ||
+        PSA_ALG_IS_PAKE(cipher_suite->algorithm ) == 0 ||
+        ( cipher_suite->type != PSA_PAKE_PRIMITIVE_TYPE_ECC &&
+          cipher_suite->type != PSA_PAKE_PRIMITIVE_TYPE_DH ) ||
+        PSA_ALG_IS_HASH( cipher_suite->hash ) == 0 )
+    {
+        return( PSA_ERROR_INVALID_ARGUMENT );
+    }
+
+    ;
+    memset( &operation->data.inputs, 0, sizeof( operation->data.inputs ) );
+
+    operation->data.inputs.alg = cipher_suite->algorithm;
+    operation->data.inputs.cipher_suite = *cipher_suite;
+
+    return( PSA_SUCCESS );
+}
+
+psa_status_t psa_pake_set_password_key(
+    psa_pake_operation_t *operation,
+    mbedtls_svc_key_id_t password )
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_status_t unlock_status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_key_slot_t *slot = NULL;
+
+    if( operation->stage != PSA_PAKE_OPERATION_STAGE_COLLECT_INPUTS )
+        return( PSA_ERROR_BAD_STATE );
+
+    if( operation->data.inputs.alg == PSA_ALG_NONE )
+        return( PSA_ERROR_BAD_STATE );
+
+    status = psa_get_and_lock_key_slot_with_policy( password, &slot,
+                                                    PSA_KEY_USAGE_DERIVE,
+                                                    PSA_ALG_JPAKE );
+    if( status != PSA_SUCCESS )
+        return( status );
+
+    psa_key_attributes_t attributes = {
+      .core = slot->attr
+    };
+
+    psa_key_type_t type = psa_get_key_type( &attributes );
+    psa_key_usage_t usage = psa_get_key_usage_flags( &attributes );
+
+    if( type != PSA_KEY_TYPE_PASSWORD &&
+        type != PSA_KEY_TYPE_PASSWORD_HASH )
+    {
+        status = PSA_ERROR_INVALID_ARGUMENT;
+        goto error;
+    }
+
+    if( ( usage & PSA_KEY_USAGE_DERIVE ) == 0 ) {
```

Comment:
Isn't this checked by `psa_get_and_lock_key_slot_with_policy()` above?

Reply:
Yes it seems that this can be removed after changing `psa_get_and_lock_key_slot()` -> `psa_get_and_lock_key_slot_with_policy()`. This is leftover after rebase.